x-pack/plugin/core/src/test/java/org/elasticsearch/license/XPackLicenseStateTests.java

```diff
@@ -201,246 +200,44 @@ public void testMonitoringAckNotBasicToBasic() {
         assertAckMessages(XPackField.MONITORING, from, BASIC, 2);
     }
 
-    public void testMonitoringUpdateRetention() {
-        assertAllowed(STANDARD, true, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), true);
-        assertAllowed(GOLD, true, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), true);
-        assertAllowed(PLATINUM, true, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), true);
-        assertAllowed(TRIAL, true, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), true);
-        assertAllowed(BASIC, true, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), false);
-        assertAllowed(MISSING, false, s -> s.checkFeature(Feature.MONITORING_UPDATE_RETENTION), false);
-    }
-
-    public void testWatcherPlatinumGoldTrialStandard() throws Exception {
-        assertAllowed(TRIAL, true, s -> s.checkFeature(Feature.WATCHER), true);
-        assertAllowed(GOLD, true, s -> s.checkFeature(Feature.WATCHER), true);
-        assertAllowed(PLATINUM, true, s -> s.checkFeature(Feature.WATCHER), true);
-        assertAllowed(STANDARD, true, s -> s.checkFeature(Feature.WATCHER), true);
-    }
-
-    public void testWatcherBasicLicense() throws Exception {
-        assertAllowed(BASIC, true, s -> s.checkFeature(Feature.WATCHER), false);
-    }
-
-    public void testWatcherInactive() {
-        assertAllowed(BASIC, false, s -> s.checkFeature(Feature.WATCHER), false);
-    }
-
-    public void testWatcherInactivePlatinumGoldTrial() throws Exception {
-        assertAllowed(TRIAL, false, s -> s.checkFeature(Feature.WATCHER), false);
-        assertAllowed(GOLD, false, s -> s.checkFeature(Feature.WATCHER), false);
-        assertAllowed(PLATINUM, false, s -> s.checkFeature(Feature.WATCHER), false);
-        assertAllowed(STANDARD, false, s -> s.checkFeature(Feature.WATCHER), false);
```

Comment:
The new tests do not cover the scenario of expired licenses. Do we need to add them with `LicensedFeature.momentaryLenient`? I know we plan to tighten the license expiration behaviour. So maybe it is not necessary?

Reply:
Sure, I can add a test for this, at least at a generic level.